experimental/SetPartitions/src/LinearPartition.jl

```diff
@@ -0,0 +1,201 @@
+"""
+    LinearPartition{S <: AbstractPartition, T <: RingElem}
+
+`LinearPartition` represents a linear combination of partitions of type `S`
+with coefficients of type `T`. 
+
+See for example Chapter 5 in [Gro20](@cite) for further information on linear 
+combinations of partitions.
+"""
+struct LinearPartition{S <: AbstractPartition, T <: RingElem}
+    base_ring :: Ring  # parent_type(T)
+    coefficients :: Dict{S, T}
+
+    function LinearPartition{S, T}(ring::Ring, coeffs::Dict{S, T}) where {S <: AbstractPartition, T <: RingElem}
+        @req isconcretetype(S) "Linear combinations are only defined for concrete subtypes of AbstractPartition"
+        @req (typeof(ring) == parent_type(T)) "Ring type is not the parent of the coefficient type"
+        for (_, c) in coeffs
```

Comment:
```suggestion
        for c in values(coeff)
```
I think this exists and is more efficient